app/jobs/fetch_hearing_locations_for_veterans_job.rb

```
@@ -55,29 +61,57 @@ def perform
         zip_code: veteran.zip_code
       )
 
-      facility_ids = facility_ids_for_veteran(veteran)
+      facility_ids = facility_ids_for_veteran(veteran, lat, long)
 
       create_available_locations_for_veteran(veteran, lat, long, facility_ids)
     end
   end
 
   private
 
-  def ros_hash
-    bfcorlids = veterans.pluck(:file_number).map do |file_number|
-      LegacyAppeal.convert_file_number_to_vacols(file_number)
-    end
+  def ro_facility_ids
+    @ro_facility_ids ||=
+      RegionalOffice::CITIES.values.reject { |ro| ro[:facility_locator_id].nil? }.pluck(:facility_locator_id)
+  end
 
-    VACOLS::Case.where(bfcorlid: bfcorlids).pluck(:bfcorlid, :bfregoff).to_h
+  def bfcorlid_to_ro_hash
+    @bfcorlid_to_ro_hash ||= begin
+      bfcorlids = veterans.pluck(:file_number).map do |file_number|
+        LegacyAppeal.convert_file_number_to_vacols(file_number)
+      end
+
+      VACOLS::Case.where(bfcorlid: bfcorlids).pluck(:bfcorlid, :bfregoff).to_h
+    end
   end
 
-  def facility_ids_for_veteran(veteran)
-    file_number = LegacyAppeal.convert_file_number_to_vacols(veteran.file_number)
-    ro = ros_hash[file_number]
+  def facility_ids_for_veteran(veteran, lat, long)
+    ro = find_or_update_ro_for_veteran(veteran, lat, long)
 
     RegionalOffice::CITIES[ro][:alternate_locations] || [] << RegionalOffice::CITIES[ro][:facility_locator_id]
   end
 
+  def find_legacy_ro_and_update_for_veteran(veteran)
+    ro = bfcorlid_to_ro_hash[LegacyAppeal.convert_file_number_to_vacols(veteran.file_number)]
+
+    veteran.update(hearing_regional_office: ro) unless ro.nil?
+
+    ro
+  end
+
+  def fetch_and_update_ro_for_veteran(veteran, lat, long)
+    distances = VADotGovService.get_distance(lat: lat, long: long, ids: ro_facility_ids)
+
+    unless distances.empty?
```

Do we want to silently fail in this case? Seems like there should always be a return value in distances, and if there isn't we'd probably want the job to fail?

---

I was actually thinking about adding exception handling around VADotGovService so the entire job doesn't fail (thinking if it's just an issue specific to that veteran's data, we'll still cover the others). I suppose if it's a more fundamental bug, we wouldn't want to keep trying. Do you think that is a better approach?

---

We need some way to alert us that it's failing. I think at first we can start by just failing the job. Once we see how common failures are, if they're causing noise, we can report the failures to Sentry and not fail the entire job. I just want us to feel pressure to fix things if they're broken. And  broken job definitely does that :)